### PR TITLE
feat(EN-11472): add AV1 break-even calculator

### PR DESCRIPTION
**Pull request by rohrlaf**
This PR will add the new "AV1 break-even calculator" to the encoding demos: https://bitmovin.atlassian.net/browse/EN-11472

Screenshot:
image

Relates to https://github.com/bitmovin-engineering/demo-framework/pull/107

### CODE REVIEW

File: encoding/av1-roi-calculator/index.html [hunk 1]
```diff
@@ -0,0 +1,184 @@
+<p class="text-justify">
+  The purpose of this calculator is to estimate the AV1 break-even point: after how many views is it
+  worth to encode using AV1, supplementing your existing encoding.
+</p>
+
+<h2 class="my-3">Inputs</h2>
+<p class="text-justify">
+  These are price points values that may be different for different customers. Please see the reasoning
```
Comment (fuetgeo) [inline encoding/av1-roi-calculator/index.html:8]: "price point values"?

File: encoding/av1-roi-calculator/index.html [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  for the defaults on the right of the actual values. If you wish to experiment with the calculation
+  using different inputs, feel free to adjust to your liking.
```
Comment (fuetgeo) [inline encoding/av1-roi-calculator/index.html:10]: unfortunately they won't be adjustable

File: encoding/av1-roi-calculator/index.html [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+          <td>50%</td>
+          <td colspan="2">30%</td>
+        </tr>
+      </tbody>
+    </table>
+
+    <ol>
+      <li class="mb-2" style="font-size: 12px;">
+        <span id="formulaFootnote">
+          Formula break-even point: ( [AV1 encoding cost] + [AV1 ingress cost] ) / ( [CODEC egress cost] - [AV1 egress cost] )
```
Comment (fuetgeo) [inline encoding/av1-roi-calculator/index.html:178]: "egress cost" => "egress cost per view"?

File: encoding/av1-roi-calculator/js/av1-roi-calculator.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    let numberOfStreamsUhd = 2;
+    let numberOfStreamsHd = 3;
+    let numberOfStreamsSd = 4;
+
+    const multiplierStreamUhd = 4;
+    const multiplierStreamHd = 2;
+    const multiplierStreamSd = 1;
+    const multiplierTechPerTitle = 1.1;
+    const multiplierTech3Pass = 2; // Multipass
+    const multiplierCodecAv1 = 10;
```
Comment (fuetgeo) [inline encoding/av1-roi-calculator/js/av1-roi-calculator.js:63]: does this already yield the correct results?
i'm missing the H.265 codec multiplier

File: encoding/av1-roi-calculator/js/av1-roi-calculator.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+      // all stream bandwidth
+      const multiplierStreamCompositionMbps = numberOfStreamsUhd * mbpsH264Uhd
+        + numberOfStreamsHd * mbpsH264Hd
+        + numberOfStreamsSd * mbpsH264Sd;
+      const allRenditionsGbPerMinH264 = (multiplierStreamCompositionMbps * 60) / (8 * 1000); // GB (8 * 1000) per min (60 seconds)
+      const allRenditionsGbPerMinAv1 = allRenditionsGbPerMinH264 * improvementsAv1H264;
+      
+      // final result
+      const av1H264BreakEven = (encodingCostPerMinuteAv1 + allRenditionsGbPerMinAv1 * ingressCostPerGb)
+        / (egressCostPerGb * (uhdRenditionGbPerMinH264 - uhdRenditionGbPerMinAv1));
```
Comment (fuetgeo) [inline encoding/av1-roi-calculator/js/av1-roi-calculator.js:96]: we could split this up into 3 steps to make it more clear:
- calculate `oneTimeCosts`
- calculate `savingsPerView`
- divide them

**Review by fuetgeo — COMMENTED** [review]
so far, so good. just a lot of small details

File: encoding/av1-roi-calculator/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# AV1 break-even point calculation
+
+Calculate the breakeven point (ROI) of AV1 versus H264 encoding for our customers.
```
[the remark this answers is not in the document]
Comment (rohrlaf) [reply encoding/av1-roi-calculator/README.md:3]: Removed ROI from page and descriptions

File: encoding/av1-roi-calculator/js/av1-roi-calculator.js (hunk 4 above)
Comment (rohrlaf) [reply encoding/av1-roi-calculator/js/av1-roi-calculator.js:63]: I added the multipliers in the code, but then removed all unused values after implementing the calculation recursively from the spreadsheet.

File: encoding/av1-roi-calculator/index.html (hunk 2 above)
Comment (rohrlaf) [reply encoding/av1-roi-calculator/index.html]: Removed in 27c0c63

File: encoding/av1-roi-calculator/index.html [hunk 7]
```diff
@@ -0,0 +1,184 @@
+<p class="text-justify">
+  The purpose of this calculator is to estimate the AV1 break-even point: after how many views is it
+  worth to encode using AV1, supplementing your existing encoding.
```
[the remark this answers is not in the document]
Comment (rohrlaf) [reply encoding/av1-roi-calculator/index.html]: Done in 27c0c63

File: encoding/av1-roi-calculator/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (rohrlaf) [reply encoding/av1-roi-calculator/README.md]: Done in 27c0c63

File: encoding/av1-roi-calculator/js/av1-roi-calculator.js (hunk 5 above)
Comment (rohrlaf) [reply encoding/av1-roi-calculator/js/av1-roi-calculator.js]: Done in [09b786a](https://github.com/bitmovin/demos/pull/92/commits/09b786a89c475d84de1e8762d5b33095dbc9c229)

File: encoding/av1-break-even-calculator/info.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  "executable": {
+    "executable": true,
+    "indexfile": "index.html"
+  },
+  "tags": [
+    "AV1",
+    "break-even",
+    "calculator",
+    "encoding",
+    "return on investment"
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/info.json:14]: noticed that the search doesn't work on staging
not sure why

File: encoding/av1-break-even-calculator/index.html [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+                value="0.04" />
+          <div class="input-group-append"><span class="input-group-text">$</span></div>  
+        </div>
+      </div>
+    </div>
+
+    <div class="col-md-6">
+      
+      <div class="form-group">
+        <label for="numberOfStreamsUhd">Amount of UHD streams</label>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:61]:
```suggestion
        <label for="numberOfStreamsUhd">Number of UHD streams (H.264 bitrate: 12 Mbps; billing multiplier: 4)</label>
```

File: encoding/av1-break-even-calculator/index.html [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+              min="0"
+              step="1"
+              type="number"
+              value="2" />
+        <div>
+          <small>&nbsp;</small>
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="numberOfStreamsHd">Amount of HD streams</label>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:73]:
```suggestion
        <label for="numberOfStreamsHd">Number of HD streams (H.264 bitrate: 5 Mbps; billing multiplier: 2)</label>
```

File: encoding/av1-break-even-calculator/index.html [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+              min="0"
+              step="1"
+              type="number"
+              value="3" />
+        <div>
+          <small>&nbsp;</small>
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="numberOfStreamsSd">Amount of SD streams</label>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:85]:
```suggestion
        <label for="numberOfStreamsSd">Number of SD streams (H.264 bitrate: 1 Mbps; billing multiplier: 1)</label>
```

File: encoding/av1-break-even-calculator/index.html [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+    </div>
+
+  </div>
+</form>
+
+<div class="container">
+  <div class="row">
+
+    <div class="col-md-6 my-2">
+      <div>Break-even point - AV1 vs H.264</div>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:102]:
```suggestion
      <div>Break-even point - H.264+AV1 vs H.264 only</div>
```

File: encoding/av1-break-even-calculator/index.html [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+      <div id="av1-h264" class="font-weight-bold" style="font-size: 24px">--</div>
+    </div>
+     
+    <div class="col-md-6 my-2">
+      <div>Break-even point - AV1 vs H.265</div>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:107]:
```suggestion
      <div>Break-even point - H.265+AV1 vs H.265 only</div>
```

File: encoding/av1-break-even-calculator/index.html [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+          <th>H265</th>
+          <th>AV1</th>
+        </tr>
+        <tr>
+          <td>1</td>
+          <td>2</td>
+          <td>10</td>
+        </tr>
+        <tr class="table-secondary">
+          <th>AV1 improvements over H.264</th>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:162]:
```suggestion
          <th>AV1 efficiency improvement over H.264</th>
```

File: encoding/av1-break-even-calculator/index.html [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+          <th colspan="2">AV1 improvements over H.265</th>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:163]:
```suggestion
          <th colspan="2">AV1 efficiency improvement over H.265</th>
```

File: encoding/av1-break-even-calculator/info.json [hunk 16]
```diff
@@ -0,0 +1,21 @@
+{
+  "title": "AV1 break-even point calculator",
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/info.json:2]:
```suggestion
  "title": "AV1 break-even calculator",
```
short is beautiful :)

**Review by fuetgeo — COMMENTED** [review]
some more wording suggestions

File: encoding/av1-break-even-calculator/index.html [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+          <th class="col-3">UHD</th>
+        </tr>
+        <tr>
+          <td>1</td>
+          <td>2</td>
+          <td>4</td>
+        </tr>
+        <tr class="table-secondary">
+          <th>PerTitle</th>
+          <th>2-Pass</th>
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:143]: 2-pass factor is not used in the calculation
should be removed

File: encoding/av1-break-even-calculator/info.json (hunk 8 above)
Comment (rohrlaf) [reply encoding/av1-break-even-calculator/info.json:14]: Probably the search index for the static page will only be updated on a live release?

Comment (fuetgeo) [reply encoding/av1-break-even-calculator/info.json:14]: yeah, we can just try that

File: encoding/av1-break-even-calculator/index.html (hunk 17 above)
Comment (rohrlaf) [reply encoding/av1-break-even-calculator/index.html:143]: 👍

Comment (rohrlaf) [reply encoding/av1-break-even-calculator/index.html]: Removed in 2baee14

**Review by dominic-miglar — APPROVED** [review]
LGTM! :)

File: encoding/av1-break-even-calculator/index.html [hunk 18]
```diff
@@ -173,7 +173,7 @@ <h3 class="my-3">Calculation multipliers</h3>
   <ol>
     <li class="mb-2" style="font-size: 12px;">
       <span id="formulaFootnote">
-        Simplified formula break-even point: ( [AV1 encoding cost] + [AV1 ingress cost] ) / ( [CODEC egress cost per view] - [AV1 egress cost per view] )
+        Simplified formula break-even point: ( [AV1 encoding cost] + [AV1 ingress cost] ) / ( [H.26X CDN delivery cost] - [AV1 CDN delivery cost] )
```
Comment (fuetgeo) [inline encoding/av1-break-even-calculator/index.html:176]: "per view" was an important information

**Pull request merged**